### PR TITLE
NDRS-845: Added validator bonding ITS

### PR DESCRIPTION
**Pull request by darthsiroftardis**
REF: https://casperlabs.atlassian.net/browse/NDRS-845

CHANGELOG:

- Added an integration test that simulates a scenario where a sixth node in a 5 node network bonds in and joins the network
- The test will check if the sixth node is a validator by querying the auction contract and checking for the public key in the auction info

PS: This was the first time in a very very very long time I wrote in bash, so mea culpa in advance for not following any style guides

### CODE REVIEW

**Comment by goral09** [comment]
Once https://github.com/CasperLabs/casper-node/pull/1064 merges, most of the code will go away. I will wait with my review for that.

**Comment by goral09** [comment]
@darthsiroftardis You can merge with `dev` now.

**Comment by darthsiroftardis** [comment]
@goral09 Merged with `dev` , scenario can be reviewed now

File: utils/nctl/sh/scenarios/bond_its.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    assert_new_bonded_validator "6"
+
+    log "------------------------------------------------------------"
+    log "Scenario bonding complete"
+    log "------------------------------------------------------------"
+
+}
+
+
+function do_await_eras_to_bid() {
```
Comment (goral09) [inline utils/nctl/sh/scenarios/bond_its.sh:41]: Unused.

File: utils/nctl/sh/scenarios/bond_its.sh [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    source "$NCTL"/sh/contracts-auction/do_bid.sh \
+            node="$NODE_ID" \
+            amount="$BID_AMOUNT" \
+            rate="$BID_DELEGATION_RATE" \
+            quiet="TRUE"
+
+    log "node-$NODE_ID auction bid submitted -> $BID_AMOUNT CSPR"
+
+    log_step "awaiting 10 seconds for auction bid deploys to finalise"
```
Comment (goral09) [inline utils/nctl/sh/scenarios/bond_its.sh:62]: I don't think it's reliable. Better to wait somewhere for the whole era.

Comment (darthsiroftardis) [reply utils/nctl/sh/scenarios/bond_its.sh:62]: I believe I wait for an era to change once the bid has been sent

Comment (darthsiroftardis) [reply utils/nctl/sh/scenarios/bond_its.sh:62]: Yep, I wait an era change before starting the sixth node, should I increase the delay

**Review by goral09 — CHANGES_REQUESTED** [review]
We should verify that the newly bonded validator was a block's creator at some point. We need a function that will either consume the event stream or simply parse the logs in search for finalized block entry with `creator` equal to the validator's public key. @TomVasile will need a similar function in this test so you can coordinate on that.

**Comment by goral09** [comment]
@darthsiroftardis , please wait for Tom's PR to get merged to `dev` and re-open your with cleaner git history and diff.

**Comment by darthsiroftardis** [comment]
@goral09 I agree

**Comment by darthsiroftardis** [comment]
Closing in favor of https://github.com/CasperLabs/casper-node/pull/1185

**Pull request closed without merging**